src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts

```diff
@@ -80,12 +80,12 @@ export class DatasetFlowDetailsHelpers {
     }
 
     public static flowOutcomeOptions(outcome: MaybeNullOrUndefined<FlowOutcome>): { icon: string; class: string } {
-        switch (outcome) {
-            case FlowOutcome.Success:
+        switch (outcome?.__typename) {
```

Comment:
I think `flowOutcomeOptions` may accept simply `FlowOutcome` without `null` or `undefined`.
The outcome is surely available in `Finished` state only, and you should rather assert it is present at higher level, than here.

Reply:
Fixed